src/web/src/pages/CourseScheduler.vue

```diff
@@ -340,10 +355,25 @@ export default {
         this._addCourseSection(course, section);
       } catch (err) {
         if (err.type === "Schedule Conflict") {
+          //Compute Existing Course That Caused Conflict.
+          const existCourse = this.courses.find(function (currentCourse) {
+            //Search For Desired CRN In Current Course.
+            var isFound = currentCourse.sections.find(function (
+              currentSection
+            ) {
+              return currentSection.crn == err.existingSession.crn;
+            });
+            //Found Course w/ Appropriate CRN.
+            if (isFound) {
+              return currentCourse;
+            }
+          });
+          //Call Helper Function To Actually Notify Client.
```

Comment:
This seems like a replica from the lines 277-290, lets make a method for this to remove a bit of duplication.

Reply:
Agree, this can be extracted to `util.js`